### DISCUSSION TITLE
6.0.0-rc.2

### DISCUSSION THREAD

**Discussion by arturcic** [GitTools/GitVersion#4097] [Announcements]
<hr /><em>This discussion was created from the release <a href='https://github.com/GitTools/GitVersion/releases/tag/6.0.0-rc.2'>6.0.0-rc.2</a>.</em>

**Comment by 0000day** [comment]
@arturcic The latest version [6.0.0-rc.2](https://github.com/GitTools/GitVersion/releases/tag/6.0.0-rc.2) gives me:
`
Could not build the configuration instance because following exception occurred: 'Requested value 'Mainline' was not found.' Please ensure that the /overrideconfig parameters are correct and the configuration file is in the correct format.
`

My `GitVersion.yml` only has 1 line:
`mode: Mainline` (all cases tested)

Removing the line gives me expected output.
Passing it as param in the cli directly leads to the same error.
Happens on feature and main branch

Have I missed a change/doc for that param?

_Moved my comment over from the nullreference-exception issue_

**Reply by HHobeck** [reply]
Like mentioned in the breaking changes documentation:

> The Mainline mode and the related implementation has been removed completely. The new Mainline version strategy should be used instead.

and in the configuration documentation:

> The following supported workflow configurations are available in GitVersion and can be referenced by the workflow property:
> 
> GitFlow (GitFlow/v1)
> GitHubFlow (GitHubFlow/v1)
> TrunkBased (TrunkBased/preview1)

The preview built-in configuration (experimental usage only) for the TrunkBased workflow uses the Mainline version strategy:

```yaml
workflow: TrunkBased/preview1
```

Edit: Alternatively, you can use the GitHubFlow workflow with mainline version strategy as follows (experimental use only):
```yaml
workflow: GitHubFlow/v1
strategies:
- ConfiguredNextVersion
- Mainline
```

Link 1: [BREAKING_CHANGES.md](https://github.com/GitTools/GitVersion/blob/main/BREAKING_CHANGES.md)
Link 2: [configuration.md](https://github.com/GitTools/GitVersion/blob/main/docs/input/docs/reference/configuration.md)
